docs: expand community docs: contributor - recommended tools (#1777)
asyncapi/community: docs/onboarding-guide/recommended_tools.md

---
title: Recommended tools for contributors
weight: 150
---

# Contributor - Recommended Tools

> This article covers the recommended tools you should have as a contributor. With these tools, you can set up workflows and begin making contributions.
> 
> These are the recommended tools:-

Reviewer: ```suggestion
```

> - [Git](https://git-scm.com)- to work with the organization's repositories.

Reviewer: Lists must be consistent in formatting/styling. Overall, this list is consistent except for the first two bullets - you have the following format: a tool name at the beginning and description of its purpose - `<tool> to <purpose>`

Pay attention that this bullet still has an inconsistent dash symbol after the link (which I highlighted in [my previous suggestion](https://github.com/asyncapi/community/pull/1777#discussion_r1992316960))

> - [A GitHub account](https://github.com). AsyncAPI hosts all its project source code and dependencies on GitHub. You'll need a GitHub account to create issues, fork the repository, submit pull requests, and more. If you're new to GitHub, familiarize yourself with [basic GitHub functionalities and workflows](https://docs.github.com/en/get-started).

Reviewer: This bullet is also slightly inconsistent, because it has a full stop after the tool name instead of describing its purpose. At first, I felt that it is ok, since the phrasing is a bit long, so I didn't mention it in suggestions. But now I see that it can be simplified, like this: 

```suggestion
- [A GitHub account](https://github.com) to fork AsyncAPI repositories, create issues, submit pull requests, and participate in discussions. If you're new to GitHub, familiarize yourself with [basic GitHub functionalities and workflows](https://docs.github.com/en/get-started).
```

- A code editor, such as [VS Code](https://code.visualstudio.com) to work on your changes.
- Command-line terminal, such as the one included in your OS, or the terminal integrated in your code editor or IDE, to run commands.
- [Node.js and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) to build and run the website locally.
- A web browser, such as Google Chrome, Firefox or Microsoft Edge to preview your changes. 

> - Markdown All-In-One VS Code extension to preview markdown documents.

Reviewer: This line contradicts with a free choice of code editor (line 13).

Furthermore, VS Code has powerful out-of-the-box capabilities to work with Markdown documents, including preview., no extension needed. So, I would just mention something like "To preview minor changes in Markdown documents without spinning up a dev server, you can use out-of-the-box capabilities of your code editor, or install the corresponding extension. For more information, please consult your code editor's documentation."

> - To preview minor changes in Markdown documents without spinning up a dev server, you can use out-of-the-box capabilities of your code editor, or install the corresponding extension. For more information, please consult your code editor's documentation.

Reviewer: Please double check my previous suggestions on updated formatting of this list. I don't see them implemented.

Author: am sorry, I checked previous suggestions but I can't find the one on formatting of lists. Can you please guide me to it?

Reviewer: Don't worry, just a few of them. I left a few comments with explanations. Please have a look.
